configurable_lti_consumer/configurable_lti_consumer.py

```diff
@@ -1,64 +1,128 @@
+import re
+
 from django.conf import settings
 from lti_consumer import LtiConsumerXBlock
 
-from .defaults import DEFAULT_CONFIGURABLE_LTI_CONSUMER_SETTINGS
 from .exceptions import ConfigurableLTIConsumerException
 
 
 class ConfigurableLtiConsumerXBlock(LtiConsumerXBlock):
-    def __init__(self, *args, **kwargs):
-        super(ConfigurableLtiConsumerXBlock, self).__init__(*args, **kwargs)
+    """
+
+    """
 
-        if not self._runtime.__class__.__name__ == "ImportSystem":
-            # We do not need this when importing course in studio
-            self.set_editable_field_names()
+    @property
+    def editable_fields(self):
+        """
+        Remove, from editable fields, the fields that we have already configured in settings.
+
+        The `launch_url` field should always be in editable fields so our setting configurations
+        are only overrides but not "typing" for an LTI consumer Xblock. To say it differently,
+        a specific LTI consumer XBlock reverts to the behavior of a normal XBlock as soon as
+        you change its launch url... and you can do it manually even after adding the XBlock
+        via a specific link behind the `Advanced` components button. It means that the links
+        proposed when clicking on the `Advanced` components button, are just shortcuts and
+        everything works normally and is understandable.
+        """
+        editable_fields = super(ConfigurableLtiConsumerXBlock, self).editable_fields
+        preconfigured_fields = self.get_configuration(self.launch_url).get("fields", {})
+        return (
+            field
+            for field in editable_fields
+            if field == "launch_url" or field not in preconfigured_fields
+        )
 
     @classmethod
     def get_template(cls, boilerplate):
         """
-        Returns a dict that will be used by modulestore to initialize xblock
-        fields
+        Returns a dictionary that will be used by the modulestore to initialize xblock fields.
+        We need this so that the value of our preconfigured fields is set in Mongodb when
+        creating the XBlock:
+        - this is absolutely required for the `launch_url` field because it is used to associate
+          the xblock with a configuration,
+        - this is still desired for other preconfigured fields because it ensures that the xblock
+          will look the same if the course is exported and re-imported to an instance of Open edX
+          on which the configurable XBlock module is not installed. Note that if the values are
+          modified in settings after running it for a while, the existing XBlock will still hold
+          in Mongodb the values that were configured at the time of their creation.
         """
```

Comment:
I propose to replace this last point by:
Note that once XBlock is initialized with settings and added to a course, changing settings will have no effect on this instance (except for LTI credentials) as the values are store in Mongodb

Reply:
As discussed, I will split the notion of default values and of hidden fields.
Whether we return the values stored in Mongodb or the version in our settings is not engaging as we can always switch back to the values in database. So I propose to keep the current solution as it is more flexible and doesn't remove the possibility to degrade to the other solution later...